### PR TITLE
a number of query-reducing measures to enhance the performance 

### PR DESCRIPTION
**Pull request by jkozar2**
These commits contain rewritten Export Attendance sheet code and a number of query-reducing measures to enhance the performance of Attendance as it loads various pages.

### CODE REVIEW

File: tool/src/java/org/sakaiproject/attendance/tool/pages/ExportPage.java [hunk 1]
```diff
@@ -115,235 +112,140 @@ public void onSubmit() {
...
+        FileOutputStream fos = null;
+        try {
+            fos = new FileOutputStream(tempFile);
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        }
+        try {
+            wb.write(fos);
             fos.close();
             wb.close();
```
Comment (ottenhoff) [inline tool/src/java/org/sakaiproject/attendance/tool/pages/ExportPage.java]: this isn't guaranteed to close, is it?

better to use try/with/resources pattern of try/catch/finally to be sure it's closed

Comment (jkozar2) [reply tool/src/java/org/sakaiproject/attendance/tool/pages/ExportPage.java]: I have implemented the try/catch with resources model as you described, on the next commit. Thanks!

Comment (ottenhoff) [reply tool/src/java/org/sakaiproject/attendance/tool/pages/ExportPage.java]: Should the wb.close() be in a finally instead of in the try?

Comment (jkozar2) [reply tool/src/java/org/sakaiproject/attendance/tool/pages/ExportPage.java]: Done. This required a separate try/catch to be around the writeFile method's call just above. WriteFile can Throw an IOexception for an error in closing the Workbook resource, and it catches its own IOException on an error in closing the FileOutputStream.
These changes are in the latest commit.

**Comment by jkozar2** [comment]
Will this code be merged soon?

**Pull request merged**